scipy/linalg/tests/test_batch.py

```diff
@@ -286,3 +287,16 @@ def test_rsf2cs(self, dtype, rng):
         A = get_random((2, 3, 4, 4), dtype=dtype, rng=rng)
         T, Z = linalg.schur(A)
         self.batch_test(linalg.rsf2csf, (T, Z), n_out=2)
+
+    @pytest.mark.parametrize('dtype', floating)
+    def test_cholesky_banded(self, dtype, rng):
+        N, n_b, n_a, batch_shape = 6, 2, 2, (5, 3)
+        A = get_nearly_hermitian((*batch_shape, N, N), dtype=dtype,
+                                 rng=rng, atol=0)  # exactly hermitian
+        A[..., range(N), range(N)] = N
+        A[..., *np.tril_indices(N, -n_b - 1)] = 0
```

Comment:
The test fails currently with an error about the second principle minor not being positive definite. This confused me slightly as a Hermitian strictly diagonally dominant matrix with real positive diagonal entries is positive definite.

Reply:
I'll try to debug, but `_to_banded` puts a general matrix in a banded form, whereas `cholesky_banded` expected only the upper or lower banded form. Is that accounted for? It might be simplest to just create the matrix in the required form directly. Any wide matrix with the last row much larger than the rest (to make it diagonally dominant in the upper form) will work, I think. It will ignore the entries in the top left corner, so they don't have to be zeroed.

Reply:
Ah yes I suspect it is an with my use of `_to_banded` thanks I'll take another look

Reply:
Fixed, thanks for pointing me in the right direction